vortex-alp/src/array.rs

```diff
@@ -43,120 +35,88 @@ impl ALPArray {
             }
             d => vortex_bail!(MismatchedTypes: "int32 or int64", d),
         };
-        Ok(Self {
-            encoded,
-            exponents,
-            patches,
-            dtype,
-            stats: Arc::new(RwLock::new(StatsSet::new())),
-        })
-    }
 
-    pub fn encode(array: &dyn Array) -> VortexResult<ArrayRef> {
-        match ArrayKind::from(array) {
-            ArrayKind::Primitive(p) => Ok(alp_encode(p)?.into_array()),
-            _ => Err(vortex_err!("ALP can only encoding primitive arrays")),
+        let mut children = vec![];
```

Comment:
You can pre-allocate the upper bound capacity, `Vec::with_capacity(2)`

I'm sure it doesn't make a different with this few elements, but hey ho